CMakeLists.txt

```diff
@@ -0,0 +1,185 @@
+                sol/compatibility/5.x.x.inl
+                sol/compatibility/version.hpp
+                sol/compatibility.hpp
+                sol/container_traits.hpp
+                sol/container_usertype_metatable.hpp
+                sol/coroutine.hpp
+                sol/debug.hpp
+                sol/demangle.hpp
+                sol/deprecate.hpp
+                sol/environment.hpp
+                sol/error.hpp
+                sol/feature_test.hpp
+                sol/filters.hpp
+                sol/forward_detail.hpp
+                sol/forward.hpp
+                sol/function.hpp
+                sol/function_result.hpp
+                sol/function_types_core.hpp
+                sol/function_types.hpp
+                sol/function_types_overloaded.hpp
+                sol/function_types_stateful.hpp
+                sol/function_types_stateless.hpp
+                sol/function_types_templated.hpp
+                sol/inheritance.hpp
+                sol/in_place.hpp
+                sol/load_result.hpp
+                sol/object_base.hpp
+                sol/object.hpp
+                sol/optional.hpp
+                sol/optional_implementation.hpp
+                sol/overload.hpp
+                sol/property.hpp
+                sol/protected_function.hpp
+                sol/protected_function_result.hpp
+                sol/protect.hpp
+                sol/proxy_base.hpp
+                sol/proxy.hpp
+                sol/raii.hpp
+                sol/reference.hpp
+                sol/resolve.hpp
+                sol/simple_usertype_metatable.hpp
+                sol/stack_check_get.hpp
+                sol/stack_check.hpp
+                sol/stack_core.hpp
+                sol/stack_field.hpp
+                sol/stack_get.hpp
+                sol/stack_guard.hpp
+                sol/stack.hpp
+                sol/stack_pop.hpp
+                sol/stack_probe.hpp
+                sol/stack_proxy.hpp
+                sol/stack_push.hpp
+                sol/stack_reference.hpp
+                sol/state.hpp
+                sol/state_view.hpp
+                sol/string_shim.hpp
+                sol/table_core.hpp
+                sol/table.hpp
+                sol/table_iterator.hpp
+                sol/thread.hpp
+                sol/tie.hpp
+                sol/traits.hpp
+                sol/tuple.hpp
+                sol/types.hpp
+                sol/unsafe_function.hpp
+                sol/userdata.hpp
+                sol/usertype_core.hpp
+                sol/usertype.hpp
+                sol/usertype_metatable.hpp
+                sol/usertype_traits.hpp
+                sol/variadic_args.hpp
+                sol/variadic_results.hpp
+                sol/wrapper.hpp)
+
+find_package(PythonInterp 3)
+if (PYTHONINTERP_FOUND)
+  add_custom_command(OUTPUT sol.hpp COMMAND ${PYTHON_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/single.py --output ${CMAKE_CURRENT_BINARY_DIR}/sol.hpp DEPENDS ${HEADER_SRCS})
```

Comment:
I think this command will replace the wrong sol.h it should be the one in the single folder